src/main/java/crawlercommons/mimetypes/MimeTypeDetector.java

```diff
@@ -44,25 +45,33 @@
                     (byte) 0xBF
                     };
 
+    private static final int LEADING_WHITESPACE_MAX_SKIP = 16;
+
+    private final static boolean[] spaceCharacters = new boolean[256];
+    static {
+        spaceCharacters[0x09] = true; // \t - character tabulation (ht)
+        spaceCharacters[0x0a] = true; // \n - line feed (lf)
+        spaceCharacters[0x0b] = true; // line tabulation (vt)
+        spaceCharacters[0x0c] = true; // form feed (ff)
+        spaceCharacters[0x0d] = true; // \r - carriage return (cr)
+        spaceCharacters[0x20] = true; // space
+    }
+
     private static class MimeTypeEntry {
         private String mimeType;
         private byte[] pattern;
+        private boolean allowBOM;
```

Comment:
I think just a "isText" flag, which implies both BOM and leading whitespace.

Reply:
Agreed.